lms/djangoapps/discussion_api/serializers.py

```diff
@@ -290,6 +290,7 @@ class CommentSerializer(_ContentSerializer):
     endorsed_by = serializers.SerializerMethodField()
     endorsed_by_label = serializers.SerializerMethodField()
     endorsed_at = serializers.SerializerMethodField()
+    children_count = serializers.IntegerField(read_only=True)
     children = serializers.SerializerMethodField(required=False)
```

Comment:
1. I'm still unclear on how this all hangs together.  Is "children" not required because some comments don't have children, or because some requests don't return the children?  From discussions with @wajeeha-khalid, it sounds like it is because we aren't always requesting the children:
- If that is the case, @wajeeha-khalid can you point out how and where we make calls that both load children and don't load the children?  Thanks.
- If that is not that case, I again have the question regarding why we can't just count the children?